pystachio/choice.py

```diff
@@ -0,0 +1,166 @@
+# Choice types: types that can take one of a group of selected types.
+
+import json
+
+from .base import Object
+from .basic import SimpleObject
+from .compatibility import Compatibility
+from .typing import (
+    Type,
+    TypeCheck,
+    TypeFactory,
+    TypeMetaclass
+)
+
+
+class ChoiceFactory(TypeFactory):
+    """A Pystachio type representing a value which can be one of several
```

Comment:
Please use 2 space indents as in other files in this project.  I'll see if I can get some of these checks automated with a tox target to run a linter.